tests/test_wo_with_signature.py

```diff
@@ -0,0 +1,262 @@
+			config
+		)
+		# Lookup returns tuple, first element is number of registries and
+		# second is element is lookup tag and third is list of organization ids.
+		registry_count, lookup_tag, registry_list = registry_list_instance.registry_lookup()
+		logger.info("\n Registry lookup response: registry count: {} lookup tag: {} registry list: {}\n".format(
+			registry_count, lookup_tag, registry_list
+		))
+		if (registry_count == 0):
+			logger.warn("No registries found")
+			sys.exit(1)
+		# Retrieve the fist registry details.
+		registry_retrieve_result = registry_list_instance.registry_retrieve(registry_list[0])
+		logger.info("\n Registry retrieve response: {}\n".format(
+			registry_retrieve_result
+		))
+		config["tcf"]["json_rpc_uri"] = registry_retrieve_result[0]
+
+	# Prepare worker
+	req_id = 31
+	global worker_id
+	worker_registry_instance = direct_jrpc.create_worker_registry(
+		config
+	)
+	if not worker_id:
+		worker_lookup_result = worker_registry_instance.worker_lookup(
+			worker_type=WorkerType.TEE_SGX, id=req_id
+		)
+		logger.info("\n Worker lookup response: {}\n".format(
+			json.dumps(worker_lookup_result, indent=4)
+		))
+		if "result" in worker_lookup_result and \
+			"ids" in worker_lookup_result["result"].keys():
+			if worker_lookup_result["result"]["totalCount"] != 0:
+				worker_id = worker_lookup_result["result"]["ids"][0]
+			else:
+				logger.error("ERROR: No workers found")
+				sys.exit(1)
+		else:
+			logger.error("ERROR: Failed to lookup worker")
+			sys.exit(1)
+	req_id += 1
+	worker_retrieve_result = worker_registry_instance.worker_retrieve(
+		worker_id,req_id
+	)
+	logger.info("\n Worker retrieve response: {}\n".format(
+		json.dumps(worker_retrieve_result, indent=4)
+	))
+	worker_obj.load_worker(worker_retrieve_result)
+
+	logger.info("**********Worker details Updated with Worker ID" + \
+		"*********\n%s\n", worker_id)
+
+	# Convert workloadId to hex
+	workload_id = "echo-result".encode("UTF-8").hex()
+	work_order_id = secrets.token_hex(32)
+	requester_id = secrets.token_hex(32)
+	session_iv = utility.generate_iv()
+	session_key = utility.generate_key()
+	requester_nonce = secrets.token_hex(16)
+	# Create work order
+	wo_params = WorkOrderParams(
+		work_order_id, worker_id, workload_id, requester_id,
+		session_key, session_iv, requester_nonce,
+		result_uri=" ", notify_uri=" ",
+		worker_encryption_key=worker_obj.encryption_key,
+		data_encryption_algorithm="AES-GCM-256"
+	)
+	# Add worker input data
+	wo_params.add_in_data(message)
+
+	# Encrypt work order request hash
+	wo_params.add_encrypted_request_hash()
+
+	# Sign work order
+	private_key = utility.generate_signing_keys()
+	if wo_params.add_requester_signature(private_key) == False:
+		logger.error("Work order request signing failed")
+		sys.exit(1)
+
+	# Submit work order
+	logger.info("Work order submit request : %s, \n \n ",
+        wo_params.to_string())
+	work_order_instance = direct_jrpc.create_work_order(
+		config
```

Comment:
Duplicate code - call echo client without signing option in run_tests.sh

Reply:
Sure, will delete it